### PR TITLE
Convert clinic choice page and wait time tests to RTL

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
This converts the clinic choice page tests and wait time alert tests to RTL.

I also fixed some bugs on the clinic choice page and adding a missing date page test for the validation message

## Testing done
Unit testing

## Acceptance criteria
- [ ] Build passes

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/new-appointment/components/ClinicChoicePage.jsx [hunk 1]
```diff
@@ -28,7 +28,9 @@ function getPageTitle(schema, typeOfCare) {
   const typeOfCareLabel = formatTypeOfCare(typeOfCare.name);
   let pageTitle = 'Clinic choice';
   if (schema?.properties.clinicId.enum.length === 2) {
-    pageTitle = `Make a ${typeOfCareLabel} appointment at your last clinic`;
+    pageTitle = `Make ${
+      vowelCheck(typeOfCareLabel) ? 'an' : 'a'
```
Comment (jbalboni) [inline src/applications/vaos/new-appointment/components/ClinicChoicePage.jsx:32]: Added @LanceGundersen's vowel check to the main header string, which had the same problem

File: src/applications/vaos/new-appointment/components/ClinicChoicePage.jsx [hunk 2]
```diff
@@ -78,10 +82,10 @@ export function ClinicChoicePage({
       scrollAndFocus();
       document.title = `${getPageTitle(schema, typeOfCare)} | Veterans Affairs`;
     },
-    [schema && facilityDetailsStatus !== FETCH_STATUS.loading],
+    [schemaAndFacilityReady],
```
Comment (jbalboni) [inline src/applications/vaos/new-appointment/components/ClinicChoicePage.jsx:85]: There was a linter warning about not using expressions in the dependency array, so I moved that to a variable.

File: src/applications/vaos/new-appointment/components/ClinicChoicePage.jsx [hunk 3]
```diff
@@ -93,7 +97,7 @@ export function ClinicChoicePage({
             {getPageTitle(schema, typeOfCare)}
           </h1>
           Your last {typeOfCareLabel} appointment was at{' '}
-          {clinics[0].clinicFriendlyLocationName || clinics[0].clinicName}:
+          {clinics[0].serviceName}:
```
Comment (jbalboni) [inline src/applications/vaos/new-appointment/components/ClinicChoicePage.jsx:100]: Missed this in the FHIR updates

File: src/applications/vaos/tests/new-appointment/components/DateTimeSelectPage/index.unit.spec.jsx [hunk 4]
```diff
@@ -343,4 +345,321 @@ describe('VAOS <DateTimeSelectPage>', () => {
     // Cleanup
     resetFetch();
   });
+
+  it('should show validation error if no date selected', async () => {
```
Comment (jbalboni) [inline src/applications/vaos/tests/new-appointment/components/DateTimeSelectPage/index.unit.spec.jsx:349]: Noticed this validation state wasn't covered when looking at code coverage report

**Review by vbahinwillit — COMMENTED** [review]
LGTM!

**Review by vbahinwillit — APPROVED** [review]

**Pull request merged**